common/cmake/devices.cmake

```diff
@@ -1344,8 +1346,9 @@ function(ADD_FPGA_TARGET)
   #
   # ADD_FPGA_TARGET defines a FPGA build targetting a specific board.  By
   # default input files (SOURCES, TESTBENCH_SOURCES, INPUT_IO_FILE) will be
-  # implicitly passed to ADD_FILE_TARGET.  If EXPLICIT_ADD_FILE_TARGET is
-  # supplied, this behavior is supressed.
+  # implicitly passed to ADD_FILE_TARGET. If EXPLICIT_ADD_FILE_TARGET is
+  # supplied, this behavior is supressed. When AUTO_ADD_FILE_TARGETS is specified
+  # file targets will be created only if they do not exist already.
```

Comment:
Why do we need these options? In what circumstances would you use `EXPLICIT_ADD_FILE_TARGET` verse the others?

Reply:
`AUTO_ADD_FILE_TARGETS` is useful when multiple test designs share the same source files and there is no immediate possibility for putting `CMakeLists.txt` files along with them to define targets for `EXPLICIT_ADD_FILE_TARGET`.

For QuickLogic there is a separate submodule with test designs and tests from multiple archs (qlf_k4n8, pp3) refer to them. We cannot place `CMakeLists.txt` inside that submodule and defining file targets in one arch and referring to them in another is not the best way to go as there are CMake options that allow disabling specific architectues. The new option seems to be the best solution there.

Reply:
Rather than explaining it to me, update the documentation here so that everyone knows this.

Reply:
Updated the documentation with explanation of `AUTO_ADD_FILE_TARGETS` purpose and use case